exp/views/study.py

```diff
@@ -219,6 +221,8 @@ def get_initial(self):
                 initial["structure"] = structure["exact_text"]
             else:
                 initial["structure"] = json.dumps(structure)
+        if not (type(self.object.generator) is str and self.object.generator.strip()):
```

Comment:
prefer `issubclass(type(self.object.generator), str)`

Although doesn't the model already restrict you with `TextField`? Why not just `if self.object.generator`?